data/data/gcp/variables-gcp.tf

```diff
@@ -163,3 +163,9 @@ variable "gcp_private_zone_project" {
   default = ""
   description = "Project where the private managed zone will exist."
 }
+
+variable "gcp_master_secure_boot" {
+  type = string
```

Comment:
I'd like to be able to add the shileded_instance_config only in case the user provided it
For that reason, I added the dynamic block in [data/data/gcp/cluster/master/main.tf](https://github.com/openshift/installer/pull/6698/files#diff-d2a68514e186572b544fef566ba0a25ad5c4540d4df56732d75937432d7fb486)

Reply:
I should have added more context/reasoning for my previous comment:

As a rule of thumb, I try to remove as much logic as possible from the terraform configs. If it is possible to move logic into go code, I will do that so that the terraform configs are as simple, declarative configs as possible. That said, a lot of logic still ends up in the configs; we just try to minimize it.

For this particular case, based on your comment, you want to keep three states in the context of the terraform configs:

1. User-specified enabled
2. User-specified disabled
3. Not user-specfiied: default

My comment was based on the idea that we would only have two states in the config: enabled/disabled, and we would determine the default in go code. Following this idea of _our_ determining the default behavior would have pros and cons: behavior would remain explicit, but if the default behavior changes on the cloud provider side we would need to (minimally) update our code to conform. 

If my reasoning about the three states is correct, then I think the existing code makes sense and I'm happy to keep moving it forward. Was this what you had in mind for why we're conditionally creating the dynamic block?

Reply:
That is accurate, In case the user didn't specify a configuration we should get the default behavior from the cloud provider. The dynamic block allows it + in case the default changes we don't need to update our code.